widgets/linechart/internal/axes/value.go

```diff
@@ -49,12 +54,19 @@ func (v *Value) String() string {
 // NewValue returns a new instance representing the provided value, rounding
 // the value up to the specified number of non-zero decimal places.
 func NewValue(v float64, nonZeroDecimals int) *Value {
+	return NewFormattedValue(v, nonZeroDecimals, nil)
+}
+
+// NewFormattedValue returns a new instance representing the provided value,
+// using a value formatter.
+func NewFormattedValue(v float64, nonZeroDecimals int, formatter valueFormatter) *Value {
```

Comment:
The privacy seems to be a bit mixed up here. I am guessing we had to define a separate type here to avoid circular dependencies, which is fine. However this is a public (exported from package) function which now accepts a private type that won't get documented in the exported GoDoc.

We might be changing this to variadic options in which case we can also choose to not define any type and just have an option accepting an anonymous function. This package is internal after all.

```go
func ValueFormatter(vf func(float64) string) ValueOption { ... }
```

Alternatively if you choose to keep the current approach, we should probably export and document the new type.

Reply:
I thought that `Value` and `NewValue` was only used internally in the `axes` package and I tried following the same style of public factories although it was only used only on the same package.

It seemed strange to me but when I contribute to other people/group repos I try following the repos styles and decision whenever I can (IMHO uniformity is better in most cases) although sometimes I miss data for not being familiar with the repository as happened on this one, sorry.

Reply:
No worries at all, honestly both ways of specifying the options are fine. The only concern is about not mixing public and private types in one declaration. Indeed NewValue is used from other packages as well (looks like zoom uses it). Otherwise the right choice would have been to make the functions private as well.

Thank you very much for taking the extra mile and trying to match local style, it indeed shows in your code and makes the review much easier.